lib/request.js

```diff
@@ -253,7 +253,7 @@ function getHttp2Opts (opts) {
 }
 
 function getHttpOpts (opts) {
-  const httpOpts = typeof opts.http === 'object' ? opts.http : {}
+  const httpOpts = typeof opts.http === 'object' && opts.http !== null ? opts.http : {}
```

Comment:
can you add a test for this?

Reply:
yeah definetly. seems like there are some more issues.

Reply:
I removed that check, as it seems that null is not the same result as when having an empty object. I assume that it has something to do with how agentOptions are affected.